src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocBlockTagLocationCheck.java

```diff
@@ -0,0 +1,177 @@
+ * Property {@code tags} - Specify the javadoc tags to process.
+ * Default value is {@code exception, hidden, author, deprecated, throws, version, see,
+ * serialData, param, serial, provides, serialField, uses, return, since}.
+ * </li>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations
+ * if the Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">Tight-HTML Rules</a>.
+ * Default value is {@code false}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the default check:
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocBlockTagLocation"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * &#47;**
+ *  * Escaped tag &amp;#64;version (OK)
+ *  * Plain text with {&#64;code &#64;see} (OK)
+ *  * A @custom tag (OK)
+ *  * &#64;since 1.0 (OK)
+ *  * text &#64;return (violation)
+ *  * * &#64;param (violation)
+ * +* &#64;serial (violation)
+ *  * &#64;see first (OK) &#64;see second (violation)
+ *  *&#47;
+ * public int field;
+ * </pre>
+ * <p>
+ * To configure the check to verify tokens from
+ * <a href="https://openjdk.java.net/jeps/8068562">JEP 8068562</a>:
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocBlockTagLocation"&gt;
+ *   &lt;property name="tags" value="apiNote, implSpec, implNote"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * @since 8.24
+ */
+@StatelessCheck
+public class JavadocBlockTagLocationCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_BLOCK_TAG_LOCATION = "javadoc.blockTagLocation";
+
+    /**
+     * This regexp is used to extract the javadoc tags.
+     */
+    private static final Pattern JAVADOC_BLOCK_TAG_PATTERN = Pattern.compile("\\s@(\\w+)");
+
+    /**
+     * Block tags from Java 11
+     * <a href="https://docs.oracle.com/en/java/javase/11/docs/specs/doc-comment-spec.html>
+     * Documentation Comment Specification</a>.
+     */
+    private static final String[] DEFAULT_TAGS = {
+        "author",
+        "deprecated",
+        "exception",
+        "hidden",
+        "param",
+        "provides",
+        "return",
+        "see",
+        "serial",
+        "serialData",
+        "serialField",
+        "since",
+        "throws",
+        "uses",
+        "version",
+    };
+
+    /**
+     * Specify the javadoc tags to process.
+     */
+    private Set<String> tags = Arrays.stream(DEFAULT_TAGS).collect(Collectors.toSet());
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.TEXT,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode ast) {
+        if (ast.getParent().getType() != JavadocTokenTypes.JAVADOC_INLINE_TAG) {
```

Comment:
please explain why `JAVADOC_INLINE_TAG` is not a target token ?

Reply:
There should be an exception for `{@code @tag}` and `{@literal @tag}` in the check.

Originally, it was
```
    if (!isPlainTextBock(ast.getParent())) {
      // ...
    }


    /**
     * Checks that the node is {@code @code} or {@code @literal} inline tag.
     *
     * @param node the node to check
     * @return {@code true} if node is {@code @code} or {@code @literal} inline tag.
     */
    private static boolean isPlainTextBock(DetailNode node) {
        return node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG
            && Stream.of(node.getChildren())
                .anyMatch(child -> child.getType() == JavadocTokenTypes.CODE_LITERAL
                    || child.getType() == JavadocTokenTypes.LITERAL_LITERAL);
    }
```

but the coverage tests report this as redundant check, since a `TEXT` node can be children only of a '{@code}` or `{@literal}`, see https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/JavadocParser.g4#L1175

So it becomes
```
    private static boolean isPlainTextBock(DetailNode node) {
        return node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG;
    }
```

then it was inlined.

Reply:
I still do not completely get it.
Technically you are right, but Check should not be executed on javadoc that have no tags.
Check should be executed almost the same amount of time as content has tags